src/storage/datacore/restapi/custom/api.pm

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright 2024 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+package storage::datacore::restapi::custom::api;
+use strict;
+use warnings;
+use centreon::plugins::http;
+use centreon::plugins::statefile;
+use JSON::XS;
+use centreon::plugins::misc qw(empty);
+
+sub new {
+    my ($class, %options) = @_;
+    if (!defined($options{output})) {
+        print "Class Custom: Need to specify 'output' argument.\n";
+        exit 3;
+    }
+    if (!defined($options{options})) {
+        $options{output}->add_option_msg(short_msg => "Class Custom: Need to specify 'options' argument.");
+        $options{output}->option_exit();
+    }
+    my $self = {};
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'hostname:s' => { name => 'hostname' },
+        'port:s'     => { name => 'port', default => 443 },
+        'proto:s'    => { name => 'proto', default => 'https' },
+        'timeout:s'  => { name => 'timeout' },
+        'username:s' => { name => 'username' },
+        'password:s' => { name => 'password' }
+    });
+    $self->{output} = $options{output};
+    $self->{http} = centreon::plugins::http->new(%options, default_backend => 'curl');
+
+    return $self;
+}
+
+sub set_options {
+    my ($self, %options) = @_;
+
+    $self->{option_results} = $options{option_results};
+}
+sub set_defaults {}
+
+# hostname,username and password are required options
+sub check_options {
+    my ($self, %options) = @_;
+    $self->{http}->set_options(%{$self->{option_results}});
+
+    if (centreon::plugins::misc::is_empty($self->{option_results}->{hostname})) {
+        $self->{output}->add_option_msg(short_msg => 'Please set hostname option');
+        $self->{output}->option_exit();
+    }
+    if (centreon::plugins::misc::is_empty($self->{option_results}->{username})) {
+        $self->{output}->add_option_msg(short_msg => 'Please set username option to authenticate against datacore rest api');
+        $self->{output}->option_exit();
+    }
+    if (centreon::plugins::misc::is_empty($self->{option_results}->{password})) {
+        $self->{output}->add_option_msg(short_msg => 'Please set password option to authenticate against datacore rest api');
+        $self->{output}->option_exit();
+    }
+
+}
```

Comment:
I think some functions usually in api.pm are missing here.
At least settings and often there is also a build_options_for_httplib function

Reply:
ok for settings, for `build_options_for_htplib` in keysight I don't understand : 
`check_options()` take data from  `$self->{option_results}` to add them to `$self->{}`
And then `build_options_for_htplib` take data from `$self->{}` to add them to `$self->{option_results}`, I don't see why ?